### PR TITLE
Add matplotlib to mypy

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

Adds matplotlib as dependency to mypy

### Details

I have no idea why adding matplotlib to mypy uncovered typing problems with numpy...  Might be a difference in numpy version recently instead???

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -89,12 +90,12 @@ repos:
 
   # this validates our github workflow files
   - repo: https://github.com/python-jsonschema/check-jsonschema
-    rev: 0.28.4
+    rev: 0.28.5
```
Comment (MatthewFlamm) [inline .pre-commit-config.yaml:93]:
```suggestion
    rev: 0.28.4
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -89,12 +90,12 @@ repos:
...
     hooks:
       - id: check-github-workflows
 
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.4.7
+    rev: v0.4.9
```
Comment (MatthewFlamm) [inline .pre-commit-config.yaml:98]:
```suggestion
    rev: v0.4.7
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -103,7 +104,7 @@ repos:
         exclude: tests/core/typing
 
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v3.1.0
+    rev: v4.0.0-alpha.8
```
Comment (MatthewFlamm) [inline .pre-commit-config.yaml:107]:
```suggestion
    rev: v3.1.0
```

File: pyvista/core/grid.py [hunk 4]
```diff
@@ -264,10 +264,11 @@ def meshgrid(self) -> list[NumpyArray[float]]:
 
         Returns
         -------
-        list[numpy.ndarray]
+        tuple[numpy.ndarray]
             List of numpy arrays representing the points of this mesh.
 
         """
+        # numpy doesn't type shape of tuple
```
Comment (MatthewFlamm) [inline pyvista/core/grid.py:271]:
```suggestion
```

**Comment by MatthewFlamm** [comment]
Before merging this, we need to have a more holistic decision on what to do with numpy 2.0.  I realize that this PR also mixes in some numpy 2.0 changes, I will make draft for now.

**Comment by MatthewFlamm** [comment]
Will address the numpy items in #6280 being proceeding here

**Comment by MatthewFlamm** [comment]
This PR now only includes matplotlib changes and is ready to review

**Review by tkoyama010 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**